datalake/src/test/java/com/sequenceiq/datalake/service/sdx/attach/SdxDetachNameGeneratorTest.java

```diff
@@ -0,0 +1,45 @@
+package com.sequenceiq.datalake.service.sdx.attach;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.MockitoAnnotations;
+
+public class SdxDetachNameGeneratorTest {
+    private static final String TEST_NAME = "test";
+
+    @InjectMocks
+    private SdxDetachNameGenerator sdxDetachNameGenerator;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    void testGenerateDetachedClusterName() {
```

Comment:
I think there should be an assertion for the full generated name

Reply:
The full generated name is created using the Date().getTime() method which returns the number of milliseconds since the general Unix epoch. 
So verifying the name would require mocking the response from Date().getTime() since the actual value can't be hardcoded. 

I am okay with mocking the response but figured it was not necessary for the test and would not mimic a real life result either way. The main goal of the generator is just to modify the initial string with a hyphen "-" followed by a non-deterministic set of characters.

Reply:
you can use regex to verify the format of the full name

Reply:
Thank you that makes sense, I'll do that.

Reply:
I not only added this to the test but went ahead and used regex matching in the original name generator. This should help strengthen the method overall so thank you for this advice.
Let me know if it looks okay.